enterprise/backup/src/test/java/org/neo4j/backup/BackupServiceIT.java

```diff
@@ -551,29 +548,36 @@ public void shouldContainTransactionsThatHappenDuringBackupProcess() throws Thro
             @Override
             public void run()
             {
-                awaitLatch( firstStoreFinishedStreaming );
-
-                createAndIndexNode( db, 1 );
-                db.getDependencyResolver().resolveDependency( DataSourceManager.class ).getDataSource()
-                  .getNeoStore().flush();
+                try
+                {
+                    latch.await();
+                }
+                catch ( InterruptedException e )
+                {
+                    throw new RuntimeException( e );
+                }
+                createNode( db, DynamicLabel.label( "Person" ) );
+                createAndIndexNode( db, 2 );
+                txId.set( lastTxId( db ) );
 
-                transactionCommitted.countDown();
             }
         } );
 
-        BackupService.BackupOutcome backupOutcome = backupService.doFullBackup( BACKUP_HOST, backupPort,
+        BackupService.BackupOutcome backupOutcome = backupService.doFullBackup( "localhost", backupPort,
                 backupDir.getAbsolutePath(), true, withOnlineBackupEnabled, BackupClient.BIG_READ_TIMEOUT, false );
 
         backup.stop();
+        backup.shutdown();
+
         executor.shutdown();
-        executor.awaitTermination( 30, TimeUnit.SECONDS );
+        assertTrue( executor.awaitTermination( 30, TimeUnit.SECONDS ) );
 
         // then
         assertEquals( DbRepresentation.of( db ), DbRepresentation.of( backupDir ) );
         assertTrue( backupOutcome.isConsistent() );
 
         // also verify the last committed tx id is correctly set
-        checkPreviousCommittedTxIdFromFirstLog( expectedLastTxId );
+        assertEquals( txId.get(), new NeoStoreUtil( backupDir, fileSystem ).getLastCommittedTx() );
```

Comment:
There is a race in the test case that might make this assertion fail.

There is nothing to ensure that the store copy doesn't fully complete before the thread committing the transaction even wakes up.